ls1intum/Artemis
# `Exam mode`: Fix test exam result view after the working period is over

### Checklist
#### General
#### Server

### Motivation and Context
Fixes student issue 13 from the exam mode testing. 

On test exams, students are no longer able to view their own submissions after the working period is over.

### Description
The endpoint used a method `isAllowedToGetExamResult` which does not consider test exams.

Additionally, I now remove the exam object from the response since it should not be passed to a student.

### Steps for Testing
Prerequisites:
- 1 Student
- 1 Exam, 1 Test exam

1. Participate in the test exam
2. Wait until the test exam working window is closed
3. View your submission and press on `View Result` on the summary page.
4. Verify that everything works as expected
5. Verify that the normal exam mode did not break. No results should be visible to students before the result are published.


### Review Progress

#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage
#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [AuthorizationCheckService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5186/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/AuthorizationCheckService.html) | 92% | ✅ |
| [ModelingSubmissionResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5186/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/ModelingSubmissionResource.html) | 96% | ✅ |
| [TextExerciseResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5186/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/TextExerciseResource.html) | 99% | ✅ |
